### DISCUSSION TITLE
Sonoff NSPanel

### DISCUSSION THREAD

**Discussion by PiotrMachowski** [HASwitchPlate/openHASP#227] [Hardware]
Hi,

Has anyone tried using openHASP with [Sonoff NSPanel](https://www.kickstarter.com/projects/sonoffnspanel/sonoff-nspanel-smart-scene-wall-switch)? It uses ESP32 and has been confirmed to work with ESPHome ([description](https://www.espthings.io/index.php/2021/10/10/sonoff-nspanel-first-impressions)).

**Comment by fvanroie** [comment]
Nextion displays only use RX/TX serial protocol and have a proprietary Nextion firmware running on the MCU attached to the ribbon cable of the Discovery display. ESPhome can be flashed to the ESP32 but it seems they are having trouble connecting to the Nextion display over serial:

> We did not manage to get the display working yet. We are reanonably sure GPIO 16 and 17 are used for TX/RX and since the original Sonoff firmware uses 921600bps for USART-updates, we are also quit sure the Baud Rate is correct. Nevertheless we keep getting `[W][nextion:072]: Nextion is not connected!` in the logs.

openHASP needs direct access to a _'dumb'_ commodity display to write directly to the screen. The design of the NSPanel + Nextion display makes that impossible. ESPhome or HASPone for ESP32 might still be possible. Tasmota is also looking into the device, so to be continued...

**Reply by PiotrMachowski** [reply]
>ESPhome can be flashed to the ESP32 but it seems they are having trouble connecting to the Nextion display over serial

Yeah, I have seen it and I hope they will figure it out

It's a shame that openHASP is not possible, I really wanted to use it

**Reply by fvanroie** [reply]
The purpose of openHASP was to eliminate the dependency on proprietary Nextion hardware and use open hardware designs.
I'm optimistic about the future now that Sonoff is joining the switchplate market, competitors will step up their game too.😃

**Reply by PiotrMachowski** [reply]
Sonoff has just published some technical details of NSPanel: [link](https://www.kickstarter.com/projects/sonoffnspanel/sonoff-nspanel-smart-scene-wall-switch/posts/3328417)

**Reply by fvanroie** [reply]
Interesting that there is an SPI connection to the display. I wonder if it can be controlled directly...

*edit* It seems the SPI is only used for the SD card on the [Nextion Discovery](https://nextion.tech/discovery-series-introduction/) panels:
![image](https://user-images.githubusercontent.com/15969459/137139461-8e05ed43-3123-4498-9684-f6a8aab6cce7.png)

**Comment by fvanroie** [comment]
We now know that the NSpanel hardware is not compatible with openHASP.
You can use Tasmota or ESPhome with this device.

**Reply by ivdimitro** [reply]
Hi, I see there has been a lot of development around the NSpanel and there seems to be even "sutom firmware" [https://github.com/peepshow-21/ns-flash](https://github.com/peepshow-21/ns-flash)
Not sure if this will change the compatibility of the NSpanel in any way with openHASP. 

P.S. I appreciate your work on open hardware but I need a device with high "wife acceptance factor" and I am afraid 3D printing will not deliver it :)

**Reply by fvanroie** [reply]
> Not sure if this will change the compatibility

No, the hardware is not compatible because of the **Serial** Nextion display. That's why openHASP can't use that display.

**Comment by hallard** [comment]
Sorry for this topic revival, but in your case may be this one would work

![image](https://github.com/user-attachments/assets/bbf653a7-58e2-4c05-ab07-90a2ad546610)


https://www.aliexpress.com/item/1005006622746590.html
